STM32H723ZGx with empty sketch does not compile

**Describe the bug**
Compilation error using this configuration:
![image](https://user-images.githubusercontent.com/17040286/214787414-49a973b0-79a8-4b02-9fd7-e3da79636894.png)
And output:


```
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_if.c: In function 'USBD_reenumerate':
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_if.c:159:21: error: 'USBD_USB_INSTANCE' undeclared (first use in this function)
  159 |   USB_DevDisconnect(USBD_USB_INSTANCE);
      |                     ^~~~~~~~~~~~~~~~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_if.c:159:21: note: each undeclared identifier is reported only once for each function it appears in
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_conf.c: In function 'USBD_LL_Init':
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_conf.c:494:21: error: 'USB' undeclared (first use in this function)
  494 |   g_hpcd.Instance = USB;
      |                     ^~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_conf.c:494:21: note: each undeclared identifier is reported only once for each function it appears in
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_ep_conf.c:38:16: error: 'PMA_EP0_OUT_ADDR' undeclared here (not in a function)
   38 |   {0x00,       PMA_EP0_OUT_ADDR, PCD_SNG_BUF},
      |                ^~~~~~~~~~~~~~~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_ep_conf.c:38:34: error: 'PCD_SNG_BUF' undeclared here (not in a function)
   38 |   {0x00,       PMA_EP0_OUT_ADDR, PCD_SNG_BUF},
      |                                  ^~~~~~~~~~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_ep_conf.c:39:16: error: 'PMA_EP0_IN_ADDR' undeclared here (not in a function)
   39 |   {0x80,       PMA_EP0_IN_ADDR,  PCD_SNG_BUF},
      |                ^~~~~~~~~~~~~~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_ep_conf.c:40:16: error: 'PMA_CDC_OUT_ADDR' undeclared here (not in a function)
   40 |   {CDC_OUT_EP, PMA_CDC_OUT_ADDR, PCD_DBL_BUF},
      |                ^~~~~~~~~~~~~~~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_ep_conf.c:40:34: error: 'PCD_DBL_BUF' undeclared here (not in a function)
   40 |   {CDC_OUT_EP, PMA_CDC_OUT_ADDR, PCD_DBL_BUF},
      |                                  ^~~~~~~~~~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_ep_conf.c:41:16: error: 'PMA_CDC_IN_ADDR' undeclared here (not in a function)
   41 |   {CDC_IN_EP,  PMA_CDC_IN_ADDR,  PCD_SNG_BUF},
      |                ^~~~~~~~~~~~~~~
C:\Users\FarisFachrurrobbiSil\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.4.0\cores\arduino\stm32\usb\usbd_ep_conf.c:42:16: error: 'PMA_CDC_CMD_ADDR' undeclared here (not in a function)
   42 |   {CDC_CMD_EP, PMA_CDC_CMD_ADDR, PCD_SNG_BUF}
      |                ^~~~~~~~~~~~~~~~
exit status 1
Error compiling for board Generic STM32H7 Series.
```


**To Reproduce**
1. Create empty sketch
2. Choose configuration just like in the image
3. Compile

**Expected behavior**
Compilation Success

**Desktop Env:**
 - OS: Windows
 - Arduino IDE version: 1.8.19
 - STM32 core version: 2.4.0

--- a/cores/arduino/stm32/usb/usbd_conf.h
+++ b/cores/arduino/stm32/usb/usbd_conf.h
@@ -35,6 +35,10 @@ extern "C" {
 #if defined(USE_USB_HS) && !defined(USB_OTG_HS)
 #error "This board does not support USB High Speed! Select 'Full Speed' in the 'Tools->USB interface' menu"
 #endif
+#if !defined(USB_BASE) && !defined(USB_OTG_FS) && defined(USB_OTG_HS) && !defined(USE_USB_HS)
+#error "This board support only USB High Speed! Select 'High Speed' or 'High Speed in Full Speed mode' in the 'Tools->USB interface' menu"
+#endif
+
 
 #include <stdio.h>
 #include <stdlib.h>

--- a/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/generic_clock.c
+++ b/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/generic_clock.c
@@ -24,6 +24,7 @@ WEAK void SystemClock_Config(void)
   RCC_OscInitTypeDef RCC_OscInitStruct = {};
   RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
   RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
+  RCC_CRSInitTypeDef RCC_CRSInitStruct = {};
 
   /** Supply configuration update enable
   */
@@ -89,5 +90,27 @@ WEAK void SystemClock_Config(void)
   if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInitStruct) != HAL_OK) {
     Error_Handler();
   }
+
+  /*Configure the clock recovery system (CRS)**********************************/
+
+  /*Enable CRS Clock*/
+  __HAL_RCC_CRS_CLK_ENABLE();
+
+  /* Default Synchro Signal division factor (not divided) */
+  RCC_CRSInitStruct.Prescaler = RCC_CRS_SYNC_DIV1;
+
+  /* Set the SYNCSRC[1:0] bits according to CRS_Source value */
+  RCC_CRSInitStruct.Source = RCC_CRS_SYNC_SOURCE_USB1;
+
+  /* HSI48 is synchronized with USB SOF at 1KHz rate */
+  RCC_CRSInitStruct.ReloadValue =  RCC_CRS_RELOADVALUE_DEFAULT;
+  RCC_CRSInitStruct.ErrorLimitValue = RCC_CRS_ERRORLIMIT_DEFAULT;
+  RCC_CRSInitStruct.Polarity = RCC_CRS_SYNC_POLARITY_RISING;
+
+  /* Set the TRIM[5:0] to the default value */
+  RCC_CRSInitStruct.HSI48CalibrationValue = RCC_CRS_HSI48CALIBRATION_DEFAULT;
+
+  /* Start automatic synchronization */
+  HAL_RCCEx_CRSConfig(&RCC_CRSInitStruct);
 }
 #endif /* ARDUINO_GENERIC_* */

--- a/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/generic_clock.c
+++ b/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/generic_clock.c
@@ -23,6 +23,7 @@ WEAK void SystemClock_Config(void)
 {
   RCC_OscInitTypeDef RCC_OscInitStruct = {};
   RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
+  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
 
   /** Supply configuration update enable
   */
@@ -69,6 +70,23 @@ WEAK void SystemClock_Config(void)
   RCC_ClkInitStruct.APB4CLKDivider = RCC_APB4_DIV2;
 
   if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_3) != HAL_OK) {
+    Error_Handler();
+  }
+
+  /** Initializes the peripherals clock
+  */
+  PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_SDMMC | RCC_PERIPHCLK_USB;
+  PeriphClkInitStruct.PLL2.PLL2M = 32;
+  PeriphClkInitStruct.PLL2.PLL2N = 96;
+  PeriphClkInitStruct.PLL2.PLL2P = 2;
+  PeriphClkInitStruct.PLL2.PLL2Q = 2;
+  PeriphClkInitStruct.PLL2.PLL2R = 4;
+  PeriphClkInitStruct.PLL2.PLL2RGE = RCC_PLL2VCIRANGE_1;
+  PeriphClkInitStruct.PLL2.PLL2VCOSEL = RCC_PLL2VCOWIDE;
+  PeriphClkInitStruct.PLL2.PLL2FRACN = 0.0;
+  PeriphClkInitStruct.SdmmcClockSelection = RCC_SDMMCCLKSOURCE_PLL2;
+  PeriphClkInitStruct.UsbClockSelection = RCC_USBCLKSOURCE_HSI48;
+  if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInitStruct) != HAL_OK) {
     Error_Handler();
   }
 }

--- a/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/variant_NUCLEO_H723ZG.cpp
+++ b/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/variant_NUCLEO_H723ZG.cpp
@@ -184,6 +184,7 @@ WEAK void SystemClock_Config(void)
   RCC_OscInitTypeDef RCC_OscInitStruct = {};
   RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
   RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
+  RCC_CRSInitTypeDef RCC_CRSInitStruct = {};
 
   /** Supply configuration update enable
   */
@@ -249,6 +250,27 @@ WEAK void SystemClock_Config(void)
     Error_Handler();
   }
 
+  /*Configure the clock recovery system (CRS)**********************************/
+
+  /*Enable CRS Clock*/
+  __HAL_RCC_CRS_CLK_ENABLE();
+
+  /* Default Synchro Signal division factor (not divided) */
+  RCC_CRSInitStruct.Prescaler = RCC_CRS_SYNC_DIV1;
+
+  /* Set the SYNCSRC[1:0] bits according to CRS_Source value */
+  RCC_CRSInitStruct.Source = RCC_CRS_SYNC_SOURCE_USB1;
+
+  /* HSI48 is synchronized with USB SOF at 1KHz rate */
+  RCC_CRSInitStruct.ReloadValue =  RCC_CRS_RELOADVALUE_DEFAULT;
+  RCC_CRSInitStruct.ErrorLimitValue = RCC_CRS_ERRORLIMIT_DEFAULT;
+  RCC_CRSInitStruct.Polarity = RCC_CRS_SYNC_POLARITY_RISING;
+
+  /* Set the TRIM[5:0] to the default value */
+  RCC_CRSInitStruct.HSI48CalibrationValue = RCC_CRS_HSI48CALIBRATION_DEFAULT;
+
+  /* Start automatic synchronization */
+  HAL_RCCEx_CRSConfig(&RCC_CRSInitStruct);
 }
 
 #ifdef __cplusplus

--- a/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/variant_NUCLEO_H723ZG.cpp
+++ b/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/variant_NUCLEO_H723ZG.cpp
@@ -183,6 +183,7 @@ WEAK void SystemClock_Config(void)
 {
   RCC_OscInitTypeDef RCC_OscInitStruct = {};
   RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
+  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
 
   /** Supply configuration update enable
   */
@@ -230,6 +231,24 @@ WEAK void SystemClock_Config(void)
   if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_3) != HAL_OK) {
     Error_Handler();
   }
+
+  /** Initializes the peripherals clock
+  */
+  PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_SDMMC | RCC_PERIPHCLK_USB;
+  PeriphClkInitStruct.PLL2.PLL2M = 1;
+  PeriphClkInitStruct.PLL2.PLL2N = 24;
+  PeriphClkInitStruct.PLL2.PLL2P = 2;
+  PeriphClkInitStruct.PLL2.PLL2Q = 2;
+  PeriphClkInitStruct.PLL2.PLL2R = 4;
+  PeriphClkInitStruct.PLL2.PLL2RGE = RCC_PLL2VCIRANGE_3;
+  PeriphClkInitStruct.PLL2.PLL2VCOSEL = RCC_PLL2VCOWIDE;
+  PeriphClkInitStruct.PLL2.PLL2FRACN = 0;
+  PeriphClkInitStruct.SdmmcClockSelection = RCC_SDMMCCLKSOURCE_PLL2;
+  PeriphClkInitStruct.UsbClockSelection = RCC_USBCLKSOURCE_HSI48;
+  if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInitStruct) != HAL_OK) {
+    Error_Handler();
+  }
+
 }
 
 #ifdef __cplusplus

